api/envoy/config/filter/network/redis_proxy/v2/redis_proxy.proto

```diff
@@ -48,6 +54,40 @@ message RedisProxy {
     bool enable_hashtagging = 2;
   }
 
-  // Network settings for the connection pool to the upstream cluster.
+  // Network settings for the connection pool to the upstream clusters.
   ConnPoolSettings settings = 3 [(validate.rules).message.required = true];
+
+  message PrefixRoute {
+    // String prefix that must match the beginning of the keys. Envoy will always favor the longest
+    // matches.
+    string prefix = 1 [(validate.rules).string.min_bytes = 1];
+
+    // Indicates if the prefix needs to be removed from the key when forwarded.
+    bool preserve_prefix = 2;
+
+    // Upstream cluster to forward the command to.
+    string cluster = 3;
+  }
+
+  message PrefixRoutes {
+    // List of prefix routes.
+    repeated PrefixRoute routes = 1 [(gogoproto.nullable) = false];
+
+    // Indicates that prefix matching should be case insensitive.
+    bool case_insensitive = 2;
+
+    // Optional catch-all route to forward commands that doesn't match any of the routes. The
+    // catch-all route becomes required when no routes are specified.
+    string catch_all_cluster = 3;
+  }
+
+  // List of prefixes used to separate keys from different workloads to different clusters. Any
```

Comment:
If possible can you add an example here of how matching works and how overlapping is handled? It's a bit hard to grok from this description  and example would be helpful and also help me review.